### PR TITLE
`Development`: Remove programming exercise simulation and fix server tests

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
This PR removes some functionality which was not really used and reduces the time that is needed to run server test.
It also fixes a flaky test for student exams

### Steps for Testing
Navigate through the most important programming exercise pages: instructor create a programming exercise, students participate, etc.

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
no changes

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java [hunk 1]
```diff
@@ -263,7 +263,7 @@ public ResponseEntity<StudentExam> getStudentExamForConduction(@PathVariable Lon
 
         // students can not fetch the exam until 5 minutes before the exam start, we use the same constant in the client
         if (ZonedDateTime.now().plusMinutes(EXAM_START_WAIT_TIME_MINUTES).isBefore(studentExam.getExam().getStartDate())) {
-            throw new AccessForbiddenException();
+            throw new AccessForbiddenException("Students cannot download the student exams until 5 minutes before the exam start");
```
Comment (sleiss) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:266]:
```suggestion
            throw new AccessForbiddenException("Students cannot download the student exams until " + EXAM_START_WAIT_TIME_MINUTES + " minutes before the exam start");
```

Comment (sleiss) [reply src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:266]: If we want to apply this change, we should probably also change it in the comment in line 264

**Review by ge65cer — APPROVED** [review]
Tested on TS2, did not spot any issues. Code changes look reasonable, too. 👍

File: src/test/java/de/tum/in/www1/artemis/StudentExamIntegrationTest.java [hunk 2]
```diff
@@ -670,8 +669,15 @@ public void testSubmitStudentExam() throws Exception {
...
         List<StudentExam> studentExamList = prepareStudentExamsForConduction(false);
+
+        // make sure the exam is generally accessible
+        exam2.setStartDate(ZonedDateTime.now().plusMinutes(4));
+        exam2 = examRepository.save(exam2);
+
         database.changeUser("student1");
-        var studentExamResponse = request.get("/api/courses/" + course2.getId() + "/exams/" + exam2.getId() + "/student-exams/" + studentExamList.get(0).getId() + "/conduction",
+        // Important: we have to retrieve the specific exam, because the order in the list is not guaranteed, otherwise the test will be flaky
+        var studentExam1 = studentExamList.stream().filter(s -> s.getUser().getLogin().equals("student1")).findFirst().get();
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/StudentExamIntegrationTest.java:679]:
```suggestion
        var studentExam1 = studentExamList.stream().filter(s -> "student1".equals(s.getUser().getLogin())).findFirst().get();
```

**Pull request merged**